src/xdotoolify.js

```diff
@@ -326,6 +326,36 @@ _Xdotoolify.prototype.autoType = function(selector, text, relpos, timeout) {
   }
   return this;
 };
+
+_Xdotoolify.prototype.sleepUntil = function(predicate, timeout) {
+  this.operations.push({
+    type: 'sleepUntil',
+    predicate: predicate,
+    timeout: timeout || 15000
+  });
+
+  return this;
+}
+
+const _sleepUntil = async function(predicate, timeout) {
+  let result = await predicate();
+  let time = 0;
+  const interval = 100;
+
+  while(!result) {
+    if (time > timeout) {
+      throw 'Timeout exceeded';
+    }
+
+    await _sleep(interval);
+    time = time + interval;
```

Comment:
this is not quite going to work because `_sleep` doesn't have a guaranteed accuracy, particularly on ci machines that are sometimes starved for resources.

Should be something like:

```
let expires = Date.now() + timeout;

while (!result) {
  if (Date.now() > expires) {
    throw;
  }
}
```

Reply:
Fixed